add gitlab iscan and db migration (#1)
elastio/contrib: gitlab-iscan.md

# Scan the local Repo inside your CI Using Elastio's iScan Feature  

#[Note in this example we are going to use gitlab but feel free to use any software version control tool you want]

## **1-Setup your CI environment**
### **(1) First to run elastio in any os you have to provide the credentials of the aws account holding our elastio stack so you will add the following as variables in your repo with thier values { aws_access_key_id , aws_secret_access_key,region } to secure your credintials** 
## **2-Deploy elastio inside your CI**

> ### **(1) In the CI we run the command { ``` mkdir ~/.aws && touch ~/.aws/config ~/.aws/credentials ``` } to create the folder that aws reads its credintials from

Reviewer: ```suggestion
### **(1) In the CI we run the command { ``` mkdir ~/.aws && touch ~/.aws/config ~/.aws/credentials ``` } to create the folder that AWS reads its credentials from
```

> ### **(2) then we run the second command ``````ich take the variables that we added to the repo and add them to the created  ~/.aws/config ~/.aws/credentials files

Reviewer: ```suggestion
### **(2) then we run the second command which take the variables that we added to the repo and adds them to the created  ~/.aws/config ~/.aws/credentials files
```

``` 
    - |
      cat << END_TEXT > ~/.aws/config
        [default]
        region = $region
        output = text
      END_TEXT
    - |
      cat << END_TEXT > ~/.aws/credentials
        [default]
        aws_access_key_id = $aws_access_key_id
        aws_secret_access_key = $aws_secret_access_key
      END_TEXT

```

> ### **(3) we follow the elstio docs to install elastio so we run the following commands**

Reviewer: ```suggestion
### **(3) we follow the Elastio docs to install `elastio` so we run the following commands**
```

```
- apt install unzip curl git wget -y
- /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com elastio/elastio-stack/ master/scripts/install-elastio.sh)"
- curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64 zip" -o "awscliv2.zip" && unzip awscliv2.zip && ./aws/install && apt install python3-pip -y
- pip3 install --extra-index-url=https://dl.cloudsmith.io public/elastio/public/python/simple/  elastio-shell
```

> ### **(4) we follow the elstio docs to install iscan**

Reviewer: ```suggestion
### **(4) we follow the Elastio docs to install `iscan`**
```

```
- curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/elastio/elastio-stack/master/scripts/iscan > iscan
- chmod +x iscan
- mv iscan /usr/local/bin
 ```

### **(5) we use iscan to scan the repo for any malware before building the project in the repo to make sure that our application is secure **

> #### - we have used elastio file backup function before  to backup a script named malware.py 

Reviewer: ```suggestion
#### - we have used `elastio file backup` function before  to backup a script named `malware.py`
```

```

import json
import subprocess
import os
variable = subprocess.Popen("cat iscan_malware.json",stdout=subprocess.PIPE,shell=True)
s = variable.stdout.read()
o = json.loads(s)
if o['infected'] == 0:
    print("No Threat Detected")
else:
    raise SystemExit('Warning Threat Detected ')     
          
```

> and we will restore it and run it in the pipeline and this script exams the iscan results of our repo to assure that no malware was detected before building the project to docker image  or performing any action on it

Reviewer: ```suggestion
and we will restore it and run it in the pipeline and this script examines the iscan results of our repo to ensure that no malware was detected before building the project to docker image  or performing any action on it
```
